### PR TITLE
Add servo_driver

### PR DESCRIPTION
**Pull request by dvizye**

### CODE REVIEW

File: embedded/pwm_driver/pwm_test.cc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "pwm_driver.hh"
+#include <iostream>
+#include <unistd.h> // usleep
+
+int main(void) {
+	PwmDriver driver = PwmDriver("/dev/i2c-1");
```
Comment (theicfire) [inline embedded/pwm_driver/pwm_test.cc]: indentation

File: embedded/servo_driver/servo_driver.cc [hunk 2]
```diff
@@ -13,128 +13,22 @@ Adafruit PCA9685 library: https://github.com/adafruit/Adafruit-PWM-Servo-Driver-
...
-	set_pwm_freq(100);
-	return 0;
-}
-
-void ServoDriver::reset() {
-	set_register_bit(PCA9685_MODE1, 7);
-	usleep(10000);
+int ServoDriver::init() {
+  pwm_driver_.init();
+  pwm_driver_.set_pwm_freq(333);
```
Comment (theicfire) [inline embedded/servo_driver/servo_driver.cc]: Let's comment with some of the docs to this servo

File: embedded/servo_driver/servo_driver.cc [hunk 3]
```diff
@@ -13,128 +13,22 @@ Adafruit PCA9685 library: https://github.com/adafruit/Adafruit-PWM-Servo-Driver-
...
-	write_to_register(PCA9685_PRESCALE, prescale);		//write prescale to register
-	clear_register_bit(PCA9685_MODE1, 4);				    //bring it out of sleep mode
-	usleep(5000);
-	enable_auto_increment(true);							//set the mode to auto increment
+void ServoDriver::set_percentage(int channel, int percentage) {
+  // TODO(dye): Check inputs.
+  constexpr int MIN_COUNT = 1160;
+  constexpr int MAX_COUNT = 2935;
+  const int count_range = MAX_COUNT - MIN_COUNT;
+  int clock_count = count_range * static_cast<float>(percentage) / 100 + MIN_COUNT;
```
Comment (theicfire) [inline embedded/servo_driver/servo_driver.cc]: Take out the word "clock"

File: embedded/servo_driver/servo_driver.cc [hunk 4]
```diff
@@ -13,128 +13,22 @@ Adafruit PCA9685 library: https://github.com/adafruit/Adafruit-PWM-Servo-Driver-
...
-	uint8_t prescale = std::floor(prescaleval + 0.5);
-
-	// We need to go into sleep mode before setting the prescaler
-	set_register_bit(PCA9685_MODE1, 4);					//go to sleep
-	write_to_register(PCA9685_PRESCALE, prescale);		//write prescale to register
-	clear_register_bit(PCA9685_MODE1, 4);				    //bring it out of sleep mode
-	usleep(5000);
-	enable_auto_increment(true);							//set the mode to auto increment
+void ServoDriver::set_percentage(int channel, int percentage) {
+  // TODO(dye): Check inputs.
```
Comment (theicfire) [inline embedded/servo_driver/servo_driver.cc]: TODO...

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
+include_directories(${PWM_DRIVER})
 include_directories(${SERVO_DRIVER})
 
 # the driver is a library
+add_library(pwm_driver SHARED ${PWM_DRIVER}/pwm_driver.cc)
 add_library(servo_driver SHARED ${SERVO_DRIVER}/servo_driver.cc)
 add_library(i2c SHARED ${LIB_I2C}/i2c.c)
 
 # make our executable, and link libi2c
+add_executable(pwm_test ${PWM_DRIVER}/pwm_test.cc)
```
Comment (jpanikulam) [inline CMakeLists.txt]: Add a CMakeLists.txt file in the embedded folder, and put this stuff  in it,

Then do
`add_subdirectory(embedded)` in this CMakeLists.txt file

This way we don't have to put it all in here

File: embedded/servo_driver/servo_driver.cc (hunk 4 above)
Comment (dvizye) [reply embedded/servo_driver/servo_driver.cc]: stays a todo

**Pull request merged**